lib/carto/named_maps/template.rb

```diff
@@ -151,7 +151,7 @@ def options_for_carto_and_torque_layers(layer, index, is_builder)
 
         options[:sql_wrap] = layer_options[:sql_wrap] || layer_options[:query_wrapper]
 
-        attributes, interactivity = attributes_and_interactivity(layer.infowindow, layer.tooltip)
+        attributes, interactivity = attributes_and_interactivity(layer, layer_options)
```

Comment:
If all you need from `layer_options` is existing interactivity, just add a param for that. The idea of passing explicit params is for the method to be more self-documented. The reader understands exactly what the method needs, and doesn't need to track down a generic variable like `layer_options`.

```ruby
attributes_and_interactivity(layer.infowindow, layer.tooltip, layer_options[:interactivity])
```